File: app/src/main/java/com/jamburger/kitter/components/Post.java
package com.jamburger.kitter.components;

import com.google.firebase.firestore.DocumentReference;

import java.util.ArrayList;
import java.util.List;

public class Post {
    String creator, postid, imageUrl, caption, kitt;
    List<DocumentReference> likes;
    List<Comment> comments;

    public Post(String creator, String postid, String imageUrl, String caption) {
        this.creator = creator;
        this.postid = postid;
        this.imageUrl = imageUrl;
        this.caption = caption;
        this.likes = new ArrayList<>();
        this.comments = new ArrayList<>();
        this.kitt = "";
    }

    public String getKitt() {
        return kitt;
    }

    public Post() {
    }

    public void setKitt(String kitt) {
        this.kitt = kitt;
    }

    public List<Comment> getComments() {
        return comments;
    }

    public void setComments(List<Comment> comments) {
        this.comments = comments;
    }


    public List<DocumentReference> getLikes() {
        return likes;
    }

    public void setLikes(List<DocumentReference> likes) {
        this.likes = likes;
    }


    public String getPostid() {
        return postid;
    }

    public String getCreator() {
        return creator;
    }


    public String getImageUrl() {
        return imageUrl;
    }


    public String getCaption() {
        return caption;
    }

}

File: app/src/main/java/com/jamburger/kitter/components/Post.kt
package com.jamburger.kitter.components

import com.google.firebase.firestore.DocumentReference

class Post {
    lateinit var creator: String
    lateinit var postid: String
    lateinit var imageUrl: String
    lateinit var caption: String
    lateinit var kitt: String
    lateinit var likes: MutableList<DocumentReference>

    constructor()
    constructor(creator: String, postid: String, imageUrl: String, caption: String) {
        this.creator = creator
        this.postid = postid
        this.imageUrl = imageUrl
        this.caption = caption
        this.likes = ArrayList()
        this.kitt = ""
    }
}
